src/pveclib/vec_int32_ppc.h

```diff
@@ -536,6 +543,60 @@ vec_clzw (vui32_t vra)
   return ((vui32_t) r);
 }
 
+/** \brief Vector Count Trailing Zeros word.
+ *
+ *  Count the number of trailing '0' bits (0-32) within each word
+ *  element of a 128-bit vector.
+ *
+ *  For POWER9 (PowerISA 3.0B) or later use the Vector Count Trailing
+ *  Zeros Word instruction <B>vctzw</B>. Otherwise use a sequence of
+ *  pre ISA 3.0 VMX instructions.
+ *  SIMDized count Trailing zeros inspired by:
+ *
+ *  Warren, Henry S. Jr and <I>Hacker's Delight</I>, 2nd Edition,
+ *  Addison Wesley, 2013. Chapter 5 Counting Bits, Section 5-4.
+ *
+ *  |processor|Latency|Throughput|
+ *  |--------:|:-----:|:---------|
+ *  |power8   |  6-8  | 2/cycle  |
+ *  |power9   |   3   | 2/cycle  |
+ *
+ *  @param vra 128-bit vector treated as 4 x 32-bit integer (words)
+ *  elements.
+ *  @return 128-bit vector with the Trailng Zeros count for each word
+ *  element.
+ */
+static inline vui32_t
+vec_ctzw (vui32_t vra)
+{
+  vui32_t r;
+#ifdef _ARCH_PWR9
+#if defined (vec_cnttz)
+  r = vec_cnttz (vra);
+#elif defined (__clang__)
+  r = vec_cnttz (vra);
```

Comment:
Nit: No difference in these two situations, so merge them into a single #if statement?

Reply:
ya ... this was copied from the CLZ implementations where GCC implemented both type specific (vec_vclzw) and type generic (vec_cntlz) forms. This was important as the ISA added wider (long long) integer operation, which the compiler may or may not have implemented as built-ins in altivec.h. Clang only implemented the type generic form. Bill wants to deprecate all type specific operations, eventually, but too many applications still use them ...

CTZ is new (POWER9) is enough that I can assume that the type generic form is supported across the integer types (up to long long) for any compiler supporting _ARCH_PWR9. So:

#ifdef _ARCH_PWR9
#if defined (vec_cnttz) || defined (__clang__)
  r = vec_cnttz (vra);
#else